test/fuzz/kubestellar_fuzz_test.go

```diff
@@ -0,0 +1,403 @@
+kind: StatusCollector
+metadata:
+  name: test-status-collector
+  namespace: default
+spec:
+  limit: 10
+  select:
+    - name: podName
+      def: object.metadata.name
+    - name: podStatus
+      def: object.status.phase
+`
+
+	baseBinding := `apiVersion: control.kubestellar.io/v1alpha1
+kind: Binding
+metadata:
+  name: test-binding
+  namespace: default
+spec:
+  bindingPolicyRef: test-binding-policy
+  workloadSelector:
+    apiGroup: ""
+    resource: "pods"
+    namespace: "default"
+    name: "test-pod"
+`
+
+	// Add seed corpus
+	f.Add([]byte(baseBindingPolicy))
+	f.Add([]byte(baseStatusCollector))
+	f.Add([]byte(baseBinding))
+	f.Add([]byte(baseBindingPolicy + "---\n" + baseStatusCollector))
+	f.Add([]byte(baseBindingPolicy + "---\n" + baseBinding))
+
+	f.Fuzz(func(t *testing.T, yamlData []byte) {
+		// Skip empty data
+		if len(yamlData) == 0 {
+			return
+		}
+
+		// Try to parse as YAML
+		var obj unstructured.Unstructured
+		err := yaml.Unmarshal(yamlData, &obj)
+		if err != nil {
+			// Expected for invalid YAML, just return
+			return
+		}
+
+		// Check if it's a KubeStellar CRD
+		apiVersion := obj.GetAPIVersion()
+		kind := obj.GetKind()
+
+		if !strings.Contains(apiVersion, "kubestellar.io") {
+			return
+		}
+
+		// Validate specific CRD types
+		switch kind {
+		case "BindingPolicy":
+			// Try to convert to BindingPolicy
+			var bp v1alpha1.BindingPolicy
+			err = runtime.DefaultUnstructuredConverter.FromUnstructured(obj.UnstructuredContent(), &bp)
+			if err != nil {
+				return
+			}
+
+			// Basic validation
+			if bp.Name == "" {
+				return
+			}
+
+		case "StatusCollector":
+			// Try to convert to StatusCollector
+			var sc v1alpha1.StatusCollector
+			err = runtime.DefaultUnstructuredConverter.FromUnstructured(obj.UnstructuredContent(), &sc)
+			if err != nil {
+				return
+			}
+
+			// Basic validation
+			if sc.Name == "" {
+				return
+			}
+
+		case "Binding":
+			// Try to convert to Binding
+			var binding v1alpha1.Binding
+			err = runtime.DefaultUnstructuredConverter.FromUnstructured(obj.UnstructuredContent(), &binding)
+			if err != nil {
+				return
+			}
+
+			// Basic validation
+			if binding.Name == "" {
+				return
+			}
+		}
+	})
+}
+
+// FuzzLabelParsing tests label parsing functionality
+func FuzzLabelParsing(f *testing.F) {
+	// Add seed corpus for label parsing
+	seedCorpus := []string{
+		"app=test",
+		"environment=production",
+		"version=v1.0.0",
+		"tier=frontend",
+		"component=api",
+		"managed-by=kubestellar",
+		"owner=team-a",
+		"project=myapp",
+		"region=us-west-2",
+		"zone=us-west-2a",
+	}
+
+	for _, seed := range seedCorpus {
+		f.Add(seed)
+	}
+
+	f.Fuzz(func(t *testing.T, labelStr string) {
+		// Skip empty strings
+		if strings.TrimSpace(labelStr) == "" {
+			return
+		}
+
+		// Try to parse the label
+		label, err := util.SplitLabelKeyAndValue(labelStr)
+		if err != nil {
+			// Expected for invalid label formats, just return
```

Comment:
The test func should succeed only if either (a) the input is invalid and `util.SplitLabelKeyAndValue` returns an error or (b) the input is valid and `util.SplitLabelKeyAndValue` does NOT return an error.